app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/securities/EnterExportMovementReferenceNumberController.scala

```diff
@@ -16,17 +16,110 @@
+import play.api.data.Forms.nonEmptyText
+import play.api.mvc.Action
+import play.api.mvc.AnyContent
+import play.api.mvc.Result
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.config.ViewConfig
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.JourneyControllerComponents
-import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.mixins.WorkInProgressMixin
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.securities.EnterExportMovementReferenceNumberController.enterExportMovementReferenceNumberKey
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.securities.EnterExportMovementReferenceNumberController.exportMovementReferenceNumberForm
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.journeys.SecuritiesJourney
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.journeys.SecuritiesJourney.Checks.declarantOrImporterEoriMatchesUserOrHasBeenVerified
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.journeys.SecuritiesJourney.Checks.hasMRNAndDisplayDeclarationAndRfS
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.ids.MRN
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.services.ClaimService
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.securities.enter_export_movement_reference_number
 
 import scala.concurrent.ExecutionContext
 
 @Singleton
 class EnterExportMovementReferenceNumberController @Inject() (
-  val jcc: JourneyControllerComponents
-)(implicit ec: ExecutionContext)
-    extends SecuritiesJourneyBaseController
-    with WorkInProgressMixin[SecuritiesJourney]
+  val jcc: JourneyControllerComponents,
+  claimService: ClaimService,
+  enterExportMovementReferenceNumberPage: enter_export_movement_reference_number
+)(implicit viewConfig: ViewConfig, ec: ExecutionContext)
+    extends SecuritiesJourneyBaseController {
+
+  final override val actionPrecondition: Option[Validate[SecuritiesJourney]] =
+    Some(
+      hasMRNAndDisplayDeclarationAndRfS &
+        declarantOrImporterEoriMatchesUserOrHasBeenVerified
+    )
+  def show: Action[AnyContent]                                               = actionReadJourney { implicit request => journey =>
+    Ok(
+      enterExportMovementReferenceNumberPage(
+        exportMovementReferenceNumberForm,
+        routes.EnterExportMovementReferenceNumberController.submit()
+      )
+    ).asFuture
+  }
+
+  def submit: Action[AnyContent] = actionReadWriteJourney { implicit request => journey =>
+    exportMovementReferenceNumberForm.bindFromRequest
+      .fold(
+        formWithErrors =>
+          (
+            journey,
+            BadRequest(
+              enterExportMovementReferenceNumberPage(
+                formWithErrors.copy(data = Map.empty),
+                routes.EnterExportMovementReferenceNumberController.submit()
+              )
+            )
+          ).asFuture,
+        mrn =>
+          claimService
+            .getDisplayDeclaration(mrn)
+            .fold(
+              _ => submitMrnAndContinue(mrn, journey),
```

Comment:
shall we consider this as successful?

Reply:
I've just checked the documentation. 200 response from ACC14 will either be a 200 (Right(Some(_)) or 204 (Right(None)) from our backend. Anything else will be a 500 (Left(_)).

ACC14 is supposed to be for imports only, so we expect an error when we call it with an export MRN. if we get a successful response we show a validation error.

I will reach out to Richard to confirm

Reply:
We should get a 400 response from ACC14 'No declaration found' for the import MRN which will result in a 500 response from our backend. I'll update the code to treat 204s as errors as well